File: _daily_topic/array_matrix/1893.cpp
#include "header.h"
//检查是否区域内所有整数都被覆盖  差分数组

class Solution {
public:
    bool isCovered(vector<vector<int>>& ranges, int left, int right) {
        vector<int> diff(52, 0);
        for (auto& range : ranges) {
            ++diff[range[0]];
            --diff[range[1] + 1];
        }
        int curr = 0;
        for (int i = 1; i <= right; ++i) {
            curr += diff[i];
            if (i >= left && curr <= 0) return false;
        }
        return true;
    }
};

File: daily/array_matrix/1893.py
from typing import List
class Solution:
    def isCovered(self, ranges: List[List[int]], left: int, right: int) -> bool:
        diff = [0] * 52
        for left, right in ranges:
            diff[left] += 1
            diff[right + 1] -= 1
        curr = 0
        for i in range(1, right):
            curr += diff[i]
            if i >= left and curr <= 0:
                return False
        return True